lib/game.rb

```diff
@@ -0,0 +1,100 @@
+class Game
+  attr_accessor :board, :player_1, :player_2
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def current_player
+    self.board.turn_count % 2 == 1 ? self.player_2 : self.player_1
+  end
+
+  def over?
+    won? || draw? || full?
+  end
+
+  def won?(board = self.board.cells)
+    WIN_COMBINATIONS.each do |win_combination|
+      position_1 = board[win_combination[0]]
+      position_2 = board[win_combination[1]]
+      position_3 = board[win_combination[2]]
+
+      if (position_1 == "X" && position_2 == "X" && position_3 == "X") ||
+        (position_1 == "O" && position_2 == "O" && position_3 == "O")
+        return win_combination
+      end
+    end
+    return false
+  end
+
+  def full?(board = self.board.cells)
+    board.any? do |space|
+      if space == "" || space == " "
+        return false
+      end
+    end
+    return true
+  end
+
+  def draw?(board = self.board.cells)
+    if won?(board)
+      return false
+    elsif full?(board)
+      return true
+    else
+      return false
+    end
+  end
+
+  def winner(board = self.board.cells)
+    winning_index = won?
```

Comment:
Consider refactoring this method like so:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.